src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metric_flyout/metrics_insights_flyout.tsx

```diff
@@ -0,0 +1,130 @@
+  useEuiTheme,
+  EuiButtonEmpty,
+  EuiFlyoutFooter,
+  isDOMNode,
+  keys,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import type { MetricField } from '../../types';
+import { MetricFlyoutContent } from './metric_flyout_content';
+import { useFlyoutA11y } from '../../hooks/use_flyout_a11y';
+
+interface MetricInsightsFlyoutProps {
+  metric: MetricField;
+  esqlQuery: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const MetricInsightsFlyout = ({
+  metric,
+  esqlQuery,
+  isOpen,
+  onClose,
+}: MetricInsightsFlyoutProps) => {
+  const { euiTheme } = useEuiTheme();
+  const DEFAULT_WIDTH = euiTheme.base * 34;
+  const defaultWidth = DEFAULT_WIDTH; // Give enough room to search bar to not wrap
+  const isXlScreen = useIsWithinMinBreakpoint('xl');
+  const [flyoutWidth, setFlyoutWidth] = useLocalStorage(
+    'metricInsightsFlyout:flyoutWidth',
+    defaultWidth
+  );
+  const { a11yProps, screenReaderDescription } = useFlyoutA11y({ isXlScreen });
+
+  const onKeyDown = useCallback(
+    (ev: React.KeyboardEvent) => {
+      if (ev.target instanceof HTMLElement && ev.target.closest('.euiDataGrid__content')) {
+        // ignore events triggered from the data grid
+        return;
+      }
+
+      if (isDOMNode(ev.target) && ev.currentTarget.contains(ev.target) && ev.key === keys.ESCAPE) {
+        ev.preventDefault();
+        ev.stopPropagation();
+        onClose();
+      }
+
+      if (ev.target instanceof HTMLInputElement) {
+        // ignore events triggered from the search input
+        return;
+      }
+
+      const isTabButton = (ev.target as HTMLElement).getAttribute('role') === 'tab';
+      if (isTabButton) {
+        // ignore events triggered when the tab buttons are focused
+        return;
+      }
+
+      const isResizableButton =
+        (ev.target as HTMLElement).getAttribute('data-test-subj') === 'euiResizableButton';
+      if (isResizableButton) {
+        // ignore events triggered when the resizable button is focused
+        return;
+      }
+    },
+    [onClose]
+  );
+
+  if (!isOpen) return null;
+
+  const minWidth = euiTheme.base * 24;
+  const maxWidth = euiTheme.breakpoint.xl;
+
+  return (
+    <EuiFlyoutResizable
+      onClose={onClose}
+      type="push"
+      size={flyoutWidth}
+      onKeyDown={onKeyDown}
+      pushMinBreakpoint="xl"
+      data-test-subj="metricViewerFlyout"
+      ownFocus
+      minWidth={minWidth}
+      maxWidth={maxWidth}
+      onResize={setFlyoutWidth}
+      css={{
+        maxWidth: `${isXlScreen ? `calc(100vw - ${DEFAULT_WIDTH}px)` : '90vw'} !important`,
+      }}
+      paddingSize="m"
+      {...a11yProps}
+    >
+      {screenReaderDescription}
+      <EuiFlyoutHeader hasBorder>
+        <EuiFlexGroup alignItems="center" gutterSize="s">
+          <EuiFlexItem grow={false}>
+            <EuiText size="m">
+              <strong>Metric</strong>
```

Comment:
Discover has the following components for the title. Should we follow the same structure?

```tsx
<EuiTitle
  size="xs"
  data-test-subj="docViewerRowDetailsTitle"
  css={css`
    white-space: nowrap;
  `}
>
  <h2>{currentFlyoutTitle}</h2>
</EuiTitle>
```